File: z-clients/java/src/main/java/com/svix/diom/models/SecurityProtocol.java
// this file is @generated
package com.svix.diom.models;

import com.fasterxml.jackson.annotation.JsonValue;

public enum SecurityProtocol {
    PLAINTEXT("plaintext"),
    SSL("ssl"),
    SASL_PLAINTEXT("sasl-plaintext"),
    SASL_SSL("sasl-ssl");
    private final String value;

    SecurityProtocol(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }
}

File: z-clients/python/diom/models/security_protocol.py
# this file is @generated
from enum import Enum


class SecurityProtocol(str, Enum):
    """The connection security protocol, mapped onto librdkafka's `security.protocol`."""

    PLAINTEXT = "plaintext"
    SSL = "ssl"
    SASL_PLAINTEXT = "sasl-plaintext"
    SASL_SSL = "sasl-ssl"

    def __str__(self) -> str:
        return str(self.value)
